subprojects/robotpy-wpilib/wpilib/_impl/start.py

```diff
@@ -1,12 +1,85 @@
 import hal
 import wpilib
 import logging
+import os.path
+import sys
 import threading
 import time
 
+import importlib.metadata
+
+if sys.version_info < (3, 10):
+
+    def entry_points(group):
+        eps = importlib.metadata.entry_points()
+        return eps.get(group, [])
+
+else:
+    entry_points = importlib.metadata.entry_points
+
+
 from .report_error import reportError, reportErrorInternal
 
 
+def _log_versions():
+    import wpilib
+    import wpilib.deployinfo
+
+    import logging
+
+    data = wpilib.deployinfo.getDeployData()
+    if data:
+        logger = logging.getLogger("deploy-info")
+        logger.info(
+            "%s@%s at %s",
+            data.get("deploy-user", "<unknown>"),
+            data.get("deploy-host", "<unknown>"),
+            data.get("deploy-date", "<unknown>"),
+        )
+        if "git-hash" in data:
+            logger.info(
+                "- git info: %s (branch=%s)",
+                data.get("git-desc", "<unknown>"),
+                data.get("git-branch", "<unknown>"),
+            )
+
+    logger = logging.getLogger("wpilib")
+
+    try:
+        import robotpy.version
+    except ImportError:
+        robotpy_version = None
+    else:
+        logger.info("RobotPy version %s", robotpy.version.__version__)
+
+    logger.info("WPILib version %s", wpilib.__version__)
+
+    if wpilib.RobotBase.isSimulation():
+        logger.info("Running with simulated HAL.")
+
+        # check to see if we're on a RoboRIO
+        # NOTE: may have false positives, but it should work well enough
+        if os.path.exists("/etc/natinst/share/scs_imagemetadata.ini"):
```

Comment:
I do hope this is unreachable these days 😅

(I suppose someone could forcibly install the raspbian sim HAL wheel somehow?)